Updates to the README and install steps (#30)
mattermost/mattermost-operator: README.md

# Mattermost Operator for Kubernetes ![CircleCI branch](https://img.shields.io/circleci/project/github/mattermost/mattermost-operator/master.svg) [![Community Server](https://img.shields.io/badge/Mattermost_Community-cloud_channel-blue.svg)](https://community.mattermost.com/core/channels/cloud)

## Status

> The Mattermost operator is in alpha but being actively developed.

Reviewer: ```suggestion
The Mattermost operator is in alpha, but is being actively developed.
```
nit:

## Summary
Mattermost is a scalable, open source collaboration tool. It's written in Golang and React.

This project offers a Kubernetes Operator for Mattermost to simplify deploying and managing your Mattermost instance.

Learn more about Mattermost at https://mattermost.com.

The Mattermost server source code is available at https://github.com/mattermost/mattermost-server.

## 1. Prerequisites

### 1.1 Kubernetes Cluster with `kubectl`

> You must have a running Kubernetes 1.8.0+ cluster. If you do not, then see the [official Kubernetes documentation](https://kubernetes.io/docs/setup/) to get one set up.

Reviewer: nit: I don't think we have actively tested anything below 1.11.X, so we may want to bump this number up unless you have tried something older.

Open that up in your file editor and edit the fields under `metadata` and `spec`. The descriptions for each setting are present in that file.

Once edited to your satisfaction, save the file and run:

```bash
$ kubectl apply -f mm_clusterinstallation.yaml
```

This will install Mattermost with your desired options.

### 2.4 Uninstall Mattermost
To uninstall a Mattermost installation, simply run `kubectl delete -f` with the file you used to install Mattermost.

For example, from the AWS or Azure quick install:

```bash
$ kubectl delete -f https://raw.githubusercontent.com/mattermost/mattermost-operator/master/docs/examples/simple_aws_azure.yaml
```

For the install anywhere example:

```bash
$ kubectl delete -f https://raw.githubusercontent.com/mattermost/mattermost-operator/master/docs/examples/simple_anywhere.yaml
```

For the customized install options example:

```bash
$ kubectl delete -f mm_clusterinstallation.yaml
```

## 3. Developer flow
To test the operator locally. We recommend [Kind](https://kind.sigs.k8s.io/), however, you can use Minikube or Minishift as well.

### 3.1 Prerequisites

> To developer locally you will need the [Operator SDK](https://github.com/operator-framework/operator-sdk).

Reviewer: ```suggestion
To develop locally you will need the [Operator SDK](https://github.com/operator-framework/operator-sdk).
```

First, checkout and install the operator-sdk CLI:

```bash
$ mkdir -p $GOPATH/src/github.com/operator-framework
$ cd $GOPATH/src/github.com/operator-framework
$ git clone https://github.com/operator-framework/operator-sdk
$ cd operator-sdk
$ git checkout master
$ make dep
$ make install
```

### 3.2 Testing locally
Developing and testing local changes to Mattermost operator is fairly simple. For that you can deploy Kind and then apply the manifests to deploy the dependencies and the Mattermost operator as well.

> You dont need to push the mattermost-operator image to DockerHub or any other registry you can load the image, built with `make build-image`, directly to the Kind cluster:

Reviewer: ```suggestion
You don't need to push the mattermost-operator image to DockerHub or any other registry if testing with kind. You can load the image, built with `make build-image`, directly to the Kind cluster by running the following:
```
